### PR TITLE
groovy: no easily conflicting files in top-level (#19872)

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

Related to #19872.
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/groovy/default.nix [hunk 1]
```diff
@@ -15,8 +15,10 @@ stdenv.mkDerivation rec {
 
   installPhase = ''
     mkdir -p $out
+    mkdir -p $out/share/groovy
     rm bin/*.bat
-    mv * $out
+    mv {bin,conf,embeddable,grooid,indy,lib} $out
+    mv {licenses,LICENSE,NOTICE} $out/share/groovy
```
Comment (rycee) [inline pkgs/development/interpreters/groovy/default.nix]: Should this not be `$out/share/doc/groovy`?

Comment (rycee) [inline pkgs/development/interpreters/groovy/default.nix]: I haven't seen any explicitly stated policy on the matter but I do see some references to `share/doc` in the [Nixpkgs manual](https://nixos.org/nixpkgs/manual/).

Comment (sternenseemann) [inline pkgs/development/interpreters/groovy/default.nix]: I see.

File: pkgs/development/interpreters/groovy/default.nix [hunk 2]
```diff
@@ -15,8 +15,10 @@ stdenv.mkDerivation rec {
 
   installPhase = ''
     mkdir -p $out
+    mkdir -p $out/share/groovy
```
Comment (Mic92) [inline pkgs/development/interpreters/groovy/default.nix]: this should be `$out/share/doc/groovy` too!

**Comment by sternenseemann** [comment]
Any thoughts or changes?

**Pull request merged**